### ISSUE TITLE
Early success event from ->publish

### ISSUE THREAD

**Issue by kraih** [inway/mojo-rabbitmq-client#16]
This one took me quite some time to figure out. It appears that the object returned by `$channel->publish` emits the success event unconditionally [here](https://github.com/inway/mojo-rabbitmq-client/blob/801d36768491ff000bfd6b84e80bb82ec9d1ec7d/lib/Mojo/RabbitMQ/Client/Method/Publish.pm#L26), even before the message frame has been written to the socket.
```perl
my $message = $channel->publish(
  exchange    => 'pubsub',
  routing_key => 'anonymous.info',
  body        => 'Hello World!',
);
$message->on(success => sub {

  # This actually prevents the message from even getting written to the socket
  Mojo::IOLoop->stop;
});
$message->on(error => sub { warn pop });
$message->deliver;
```
Of course i assumed i could stop the event loop once the success event had been emitted, which is not the case. That has to be a bug, right?

**Comment by SPodjasek** [comment]
Should be fixed in 2f7958c

**Comment by kraih** [comment]
Yes, looks like that fixed it.
